### ISSUE TITLE
Problem with FluxInpaintPipeline when doing a replace.

### ISSUE THREAD

**Issue by ukaprch** [huggingface/diffusers#9486]
### Describe the bug

Since this is a relatively new feature of Flux, problems are to be expected. The images below tell the whole story.
The issue is that the black mask (the part to reserve) is overlaying the new image with a copy of the masked image.
I'm using an image of a dog sitting on a park bench. I mask the dog. I then run the inpaint for Flux with the prompt " image of a dog sitting on a swing in a zoo" and the output shows the masked image of the dog overlayed on what appears to be a total image generation. Not how replace is supposed to work. I left padding mask crop at None since this is a replace. Height and Width were set to 1024. Guidance scale = 7.5. Strength = 0.99. No latent input.
input sample:
![sample1](https://github.com/user-attachments/assets/2f40747d-82c3-4ae7-90ab-724084e99a63)
mask sample:
![imgmask_2](https://github.com/user-attachments/assets/77eedafc-4de1-4b29-9a55-96787e4ac541)
result sample (incorrect):
![imgname_2](https://github.com/user-attachments/assets/d1362338-4095-42aa-a45b-047b107d2189)
The expected output should look something like this (taken from SDXL using same inputs and prompt):
![imgname_2](https://github.com/user-attachments/assets/74a1e823-acd2-4098-9a76-0e97e1ad0077)

### Reproduction
`def replace_image_with_flux(pipe, inference_model, inference_scheduler, img, mask, seed, text_prompt, width, height, 
        guidance_scale, strength, inference_steps, device):
    start = time.perf_counter()
    dtype = torch.bfloat16
    if pipe is None:
        try:
            print('Quantizing transformer')
            class QuantizedFluxTransformer2DModel(QuantizedDiffusersModel):
                base_class = FluxTransformer2DModel
            transformer = QuantizedFluxTransformer2DModel.from_pretrained(
                "./flux-dev/fluxtransformer2dmodel_qint8"
                ).to(dtype=dtype)
            print('Quantizing text_encoder_2')
            class  QuantizedT5EncoderModelForCausalLM (QuantizedTransformersModel):
                auto_class = T5EncoderModel
                auto_class.from_config = auto_class._from_config`
`            text_encoder_2 = QuantizedT5EncoderModelForCausalLM.from_pretrained(
                "./flux-dev/t5encodermodel_qint8"
            ).to(dtype=dtype)
            pipe = FluxInpaintPipeline.from_pretrained(inference_model,token=token, transformer=transformer, text_encoder_2=text_encoder_2, torch_dtype=dtype, use_safetensors=True)
        except RuntimeError as error:
            print('Error', error)
            return None, None
    else:
        pipe = FluxInpaintPipeline.from_pipe(pipe)
    try:
        if inference_scheduler == 'None':
            pass
        elif inference_scheduler == 'FlowMatchEulerDiscrete':
            pipe.scheduler = FlowMatchEulerDiscreteScheduler.from_config(pipe.scheduler.config)
        generator = torch.Generator(device=device).manual_seed(seed)
        if device == 'cpu':
            pipe.enable_model_cpu_offload()
        else:
            pipe = pipe.to(device)
        # height and width have to be divisible by 8
        crop_size = max(height, width)
        img2arr = np.array(img)
        mask2arr = np.array(mask)
        # Calculate the aspect ratio of the image
        img_h, img_w = img2arr.shape[:2]
        aspect_ratio = float(img_w) / float(img_h)
        # If the shorter side is less than 1024, resize the image proportionally
        if min(img_h, img_w) > crop_size:
            if img_h < img_w:
                new_height = crop_size
                new_width = int(new_height * aspect_ratio)
            else:
                new_width = crop_size
                new_height = int(new_width / aspect_ratio)
            new_height = int(np.floor(new_height / 8) * 8)
            new_width = int(np.floor(new_width / 8) * 8)
            img_crop = cv2.resize(img2arr, (new_width, new_height), interpolation=cv2.INTER_LANCZOS4)
            mask_crop = cv2.resize(mask2arr, (new_width, new_height), interpolation=cv2.INTER_LANCZOS4)
        else:
            img_h = int(np.floor(img_h / 8) * 8)
            img_w = int(np.floor(img_w / 8) * 8)
            img_crop = cv2.resize(img2arr, (img_w, img_h), interpolation=cv2.INTER_LANCZOS4)
            mask_crop = cv2.resize(mask2arr, (img_w, img_h), interpolation=cv2.INTER_LANCZOS4)
        #image dimensions
        height, width, _ = img_crop.shape
        print(f'height = {height}, width = {width}')
        # print the seed
        print(f'seed = {seed}')
        #save preliminary mask
        #img_p = './results/imgmask_2.png'
        immask = Image.fromarray(255 - mask_crop)
        #immask.save(img_p)
        with torch.inference_mode():
            image = pipe(
                prompt=text_prompt,
                width=width,
                height=height,
                guidance_scale=guidance_scale,
                strength=strength,
                generator=generator,
                image=Image.fromarray(img_crop),
                mask_image=immask,           #Image.fromarray(255 - mask_crop),
                num_inference_steps=inference_steps).images[0]
        del img2arr
        del mask2arr
        del img_crop
        del mask_crop
        #save preliminary image
        img_p = './results/imgname_2.png'
        image.save(img_p)
        end = time.perf_counter()
        print (f"time: {(end - start):.2f}sec")
        pipe = pipe.to('cpu', silence_dtype_warnings=True)
        return pipe, image
    except RuntimeError as error:
        print('Error', error)
        print('If you encounter CUDA out of memory, try using device=cpu')
        return None, None`

[python.txt](https://github.com/user-attachments/files/17079128/python.txt)

### Logs

```shell
N/A
```

### System Info

Windows OS 10
Python: 3.12.5
Diffusers: diffusers-0.31.0.dev0

**Comment by asomoza** [comment]
Hi, your code is not formatted correctly so it's hard to read, but I see you do some calculations with the mask, the most noticeable one is that you're inverting it which could explain your result.

Probably best if you save the mask before you send it to the pipeline and make sure it's like the one you're showing, I suspect you invert it and it's a black subject over a white background which would make your result the expected one.

**Comment by ukaprch** [comment]
I invert the mask because I'm doing a replace (not a fill). This same
script works for SDXL inpainting so It's not my script,
but the Flux Inpainting python. Please look more deeply into this problem.

**Comment by ukaprch** [comment]
I have updated the post with more details, unfortunately the code editor doesn't seem to work very well. I have supplied what the mask would look like with the (255-mask code) and included this image as the input. Hopefully, my message is more clear.

**Comment by asomoza** [comment]
I think I now understand what you mean, so to clarify here>

- You want to preserve the dog and change the background (the mask means that).
- What you're not happy about is that the replaced background has also another dog in a park and it's not taking into account what was before in the original image as in your demo image, which shows that it preserves the dog and part of the bench.

A tested with a limited time, Flux with inpainting and img2img and it was really bad and finicky, you have a very limited range of the strength and CFG you could use to get a decent image.

Also you're probably using an inpainting model with SDXL and we don't have one with Flux, so it's even worse. You can try using differential diffusion which we have a PR that can help to take more into account the original content of the image.

Maybe @Gothos has some more insights about this.

**Comment by Gothos** [comment]
The area to-be-replaced ought to be 1s (white), other areas ought to be black.
I did see that you did do `255-mask_crop` in there though, so not super sure regd. this. Are you sure mask_crop isn't already scaled down to 1s or something like that?

**Comment by ukaprch** [comment]
I went with using the image and mask w/o downsizing/scaling as input and
still have the same problem. The problem is with the python
script pipeline_flux_inpaint.py
there's some apparent bug in there.
Here is the output I generated. As you can see the original image (masked
part black) is simply overlaid onto a new image and not blended in the way
inpainting is supposed to work.

**Comment by ukaprch** [comment]
In retrospect the problem may be a feature and not a bug in Flux inpainting. I was able to somewhat generate an image to my liking after fiddling with strength and guidance. I used strength = 0.8 and guidance = 7.5 to generate the following using the prompt: image of a dog sitting on a swing in a baseball park. I think with this in mind it would still be a good idea if a Flux Inpainting Model might help to achieve better results.

![imgname_2](https://github.com/user-attachments/assets/6e77ed38-a4fc-4ddb-9861-e04f7a071299)
